### PR TITLE
Update outdated documentation

### PR DESCRIPTION
**Pull request by yiakwy**
see commit messages for details

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+
+
+# [1.0.0](https://github.com/grpc-ecosystem/grpc-gateway/compare/v1.8.5...v1.0.0) (2019-03-28)
+
+
+
+# [1.0.0](https://github.com/grpc-ecosystem/grpc-gateway/compare/v1.8.5...v1.0.0) (2019-03-28)
+
+
+
```
Comment (johanbrandhorst) [inline CHANGELOG.md]: Remove these please

File: README.md [hunk 2]
```diff
@@ -40,9 +40,23 @@ https://github.com/protocolbuffers/protobuf/releases
 Then, `go get -u` as usual the following packages:
 
 ```sh
+go get -u google.golang.org/grpc
```
Comment (johanbrandhorst) [inline README.md]: This shouldn't be necessary to use the generators, I think? Why did you add it?

File: README.md [hunk 3]
```diff
@@ -40,9 +40,23 @@ https://github.com/protocolbuffers/protobuf/releases
...
+go get -u github.com/golang/protobu/{proto,protoc-gen-go}
+```
+In case that you have problems in using grpc golang plugin with protobuf, install it from source:
+
+```sh
+GIT_TAG="v1.2.0" # change as needed
+go get -d -u github.com/golang/protobuf/protoc-gen-go
+git -C "$(go env GOPATH)"/src/github.com/golang/protobuf checkout $GIT_TAG
+go install github.com/golang/protobuf/protoc-gen-go
+
```
Comment (johanbrandhorst) [inline README.md]: Instead of this, I think we should just use module syntax to download a specific version:
```bash
$ go get github.com/golang/protobuf/protoc-gen-go@1.2.0
```

What do you think?

File: README.md [hunk 4]
```diff
@@ -53,6 +67,8 @@ This will place three binaries in your `$GOBIN`;
 
 Make sure that your `$GOBIN` is in your `$PATH`.
 
+
+
```
Comment (johanbrandhorst) [inline README.md]: Remove this please, markdown squishes this all to one line anyway.

File: README.md [hunk 5]
```diff
@@ -102,11 +118,14 @@ annotation to your .proto file
...
+   Suppose `ROOT_TO_PROTO` is where your proto files store and `DEST` is where you want put generated files in, do:  
+
    ```sh
-   protoc -I/usr/local/include -I. \
+   protoc -I/usr/local/include \
+     -I${ROOT_TO_PROTO} \
      -I$GOPATH/src \
      -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-     --go_out=plugins=grpc:. \
+     --go_out=plugins=grpc:${DEST} \
```
Comment (johanbrandhorst) [inline README.md]: While we're doing this, we should add `paths=source_relative`. It simplifies generated file output greatly.

File: README.md [hunk 6]
```diff
@@ -102,11 +118,14 @@ annotation to your .proto file
 
 3. Generate gRPC stub
 
+   Suppose `ROOT_TO_PROTO` is where your proto files store and `DEST` is where you want put generated files in, do:  
+
    ```sh
-   protoc -I/usr/local/include -I. \
+   protoc -I/usr/local/include \
+     -I${ROOT_TO_PROTO} \
```
Comment (johanbrandhorst) [inline README.md]: I don't know if this is helping tbh. `-I.` means current directory, not `path/to/your_service.proto`. I'd say remove the `ROOT_TO_PROTO` but keep the `DEST`.

File: README.md [hunk 7]
```diff
@@ -135,10 +156,12 @@ annotation to your .proto file
 5. Generate reverse-proxy
 
    ```sh
-   protoc -I/usr/local/include -I. \
+   protoc -I/usr/local/include \
+     -I${ROOT_TO_PROTO} \
      -I$GOPATH/src \
      -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-     --grpc-gateway_out=logtostderr=true:. \
+     --plugin=proto-gen-grpc-gateway=${GOPATH}/bin/protoc-gen-grpc-gateway \
```
Comment (johanbrandhorst) [inline README.md]: Specifying the `plugin` is not necessary if it is in your path already. Using `go get` should put it in your path. I don't think we need to add this.

File: README.md [hunk 8]
```diff
@@ -135,10 +156,12 @@ annotation to your .proto file
...
+     --grpc-gateway_out=logtostderr=true,grpc_api_configuration="${ROOT_TO_service.yaml}",v=2:${DEST} \
```
Comment (johanbrandhorst) [inline README.md]: Why did you add the new parameters? This is not the standard workflow.

File: README.md [hunk 9]
```diff
@@ -193,11 +216,15 @@ annotation to your .proto file
...
+
    ```sh
-   protoc -I/usr/local/include -I. \
+   protoc -I/usr/local/include \
+     -I${ROOT_TO_PROTO} \
      -I$GOPATH/src \
      -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
-     --swagger_out=logtostderr=true:. \
+     --plugin=protoc-gen-swagger=${GOPATH}/bin/protoc-gen-swagger \
+     --swagger_out=logtostderr=true,grpc_api_configuration="${ROOT_TO_service.yaml}",v=2:${ROOT}/api/${VERSION} \
```
Comment (johanbrandhorst) [inline README.md]: Same comment as above.

File: README.md [hunk 10]
```diff
@@ -221,8 +248,13 @@ More examples are available under `examples` directory.
   * `proto/examplepb/echo_service.pb.go`, `proto/examplepb/a_bit_of_everything.pb.go`, `proto/examplepb/unannotated_echo_service.pb.go`: [generated] stub of the service
   * `proto/examplepb/echo_service.pb.gw.go`, `proto/examplepb/a_bit_of_everything.pb.gw.go`, `proto/examplepb/uannotated_echo_service.pb.gw.go`: [generated] reverse proxy for the service
   * `proto/examplepb/unannotated_echo_service.yaml`: gRPC API Configuration for ```unannotated_echo_service.proto```
-* `server/main.go`: service implementation
-* `main.go`: entrypoint of the generated reverse proxy
+*  browser/gulpfile.js: entry file in current branch to build and execute grpc server and reverse proxy gateway examples for clients to query though HTTP1.1.
+*  cmd/example-grpc-server/main.go: grpc server entry point
+*  cmd/example-grpc-gateway-server/main.go: http reverse proxy server entry point
+*  server: grpc server and servicesimplementation in golang used in this example.
```
Comment (johanbrandhorst) [inline README.md]: Missing space

File: examples/Gopkg.toml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  name = "google.golang.org/grpc"
+  version = "1.19.1"
+
+[[constraint]]
+  name = "gopkg.in/resty.v1"
+  version = "1.12.0"
+
+[prune]
+  go-tests = true
+  unused-packages = true
```
Comment (johanbrandhorst) [inline examples/Gopkg.toml]: Please remove this file. We use go modules.

File: commitlint.config.js [hunk 12]
```diff
@@ -0,0 +1,5 @@
+
+module.exports = {
+	extends: ['@commitlint/config-conventional']
+}
+
```
Comment (johanbrandhorst) [inline commitlint.config.js]: Why was this added?

File: examples/Gopkg.lock [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+    "google.golang.org/grpc",
+    "google.golang.org/grpc/codes",
+    "google.golang.org/grpc/connectivity",
+    "google.golang.org/grpc/grpclog",
+    "google.golang.org/grpc/metadata",
+    "google.golang.org/grpc/status",
+    "gopkg.in/resty.v1",
+  ]
+  solver-name = "gps-cdcl"
+  solver-version = 1
```
Comment (johanbrandhorst) [inline examples/Gopkg.lock]: Please remove this file. We use go modules.

File: examples/index.html [hunk 14]
```diff
@@ -0,0 +1,6 @@
...
+(function(){var pmc='{\x22Qnk92g\x22:{},\x22U5B21g\x22:{},\x22YFCs/g\x22:{},\x22ZI/YVQ\x22:{},\x22d\x22:{},\x22sb_he\x22:{\x22agen\x22:true,\x22cgen\x22:true,\x22client\x22:\x22heirloom-hp\x22,\x22dh\x22:true,\x22dhqt\x22:true,\x22ds\x22:\x22\x22,\x22ffql\x22:\x22en\x22,\x22fl\x22:true,\x22host\x22:\x22google.co.uk\x22,\x22isbh\x22:28,\x22jsonp\x22:true,\x22msgs\x22:{\x22cibl\x22:\x22Clear Search\x22,\x22dym\x22:\x22Did you mean:\x22,\x22lcky\x22:\x22I\\u0026#39;m Feeling Lucky\x22,\x22lml\x22:\x22Learn more\x22,\x22oskt\x22:\x22Input tools\x22,\x22psrc\x22:\x22This search was removed from your \\u003Ca href\x3d\\\x22/history\\\x22\\u003EWeb History\\u003C/a\\u003E\x22,\x22psrl\x22:\x22Remove\x22,\x22sbit\x22:\x22Search by image\x22,\x22srch\x22:\x22Google Search\x22},\x22ovr\x22:{},\x22pq\x22:\x22\x22,\x22refpd\x22:true,\x22rfs\x22:[],\x22sbpl\x22:24,\x22sbpr\x22:24,\x22scd\x22:10,\x22sce\x22:5,\x22stok\x22:\x22pYDDy2vXHPYVI_XZKtZN4GTyhCg\x22,\x22uhde\x22:false}}';google.pmc=JSON.parse(pmc);})();</script>        </body></html>
```
Comment (johanbrandhorst) [inline examples/index.html]: Where did this file come from?

File: package.json [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+      "subjectPatternErrorMsg": "Wrong commit format! Subject does not match subject pattern!",
+      "helpMessage": ""
+    }
+  },
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
+    }
+  }
+}
```
Comment (johanbrandhorst) [inline package.json]: Why did we add this? I don't think we want to publish an npm module.

File: scripts/install.sh [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+		configGitCommitter
+	
+		else
+		warn "platform <$(uname -s)> Not supported yet. Please install the package dependencies manually. Pull requests are welcome!"
+		fi
+	fi 
+
+}
+
+main
```
Comment (johanbrandhorst) [inline scripts/install.sh]: What is the purpose of this file?

File: CHANGELOG.md (hunk 1 above)
Comment (yiakwy) [reply CHANGELOG.md]: roger

File: README.md (hunk 3 above)
Comment (yiakwy) [reply README.md]: I tried protoc-gen-go@1.2.0 but it failed. In the case of future conflicts, I would like to recommend to install it from source. Many software like glog, gflags are better to be installed from source I think.

Comment (johanbrandhorst) [reply README.md]: `go get` already installs from source. Your commands are equivalent to using go get with a specific version (AFAIK).

File: README.md (hunk 5 above)
Comment (yiakwy) [reply README.md]: I think it is better to leave users to decide where to install. B.W.T, for newbee of golang like me, I didn't even realize immediately that I can change output directory. For me, I would like to put all generated files in build (I use cmake to build cpp/python/golang app) so that they are all covered by dependencies resolution system.

File: commitlint.config.js (hunk 12 above)
Comment (yiakwy) [reply commitlint.config.js]: Since you don't need my commit tools, see `scripts/install.sh`, I can delete it safely. It provides my git commit hook.

File: examples/Gopkg.lock (hunk 13 above)
Comment (yiakwy) [reply examples/Gopkg.lock]: roger

File: examples/index.html (hunk 14 above)
Comment (yiakwy) [reply examples/index.html]: I have no idea where the updates come from. We need to add it to .gitignore

File: package.json (hunk 15 above)
Comment (yiakwy) [reply package.json]: The same. It provides me git commit hook so that developers are forced to commit messages in a standard format. See git-cz.

File: scripts/install.sh (hunk 16 above)
Comment (yiakwy) [reply scripts/install.sh]: It prepares many depandencies. In this commit, it just provides me with git commit hook.

File: README.md (hunk 2 above)
Comment (yiakwy) [reply README.md]: I am not sure because I have already installed grpc cpp++ in my labtop and workstation from source before I use grpc-gateway. I think we at least need to install grpc c-core hence I add it.

File: README.md (hunk 4 above)
Comment (yiakwy) [reply README.md]: roger.

File: CHANGELOG.md (hunk 1 above)
Comment (yiakwy) [reply CHANGELOG.md]: It is automatically generated by the configured commit hook. You edit package.json for styles you want for the team.

File: README.md (hunk 10 above)
Comment (yiakwy) [reply README.md]: copy it.

File: README.md (hunk 2 above)
Comment (johanbrandhorst) [reply README.md]: No, you dont need it. Please remove it again. You need it to use the generated files, but you don't need it to generate it.

File: CHANGELOG.md (hunk 1 above)
Comment (johanbrandhorst) [reply CHANGELOG.md]: Can you please remove anything that isn't directly related to the documentation change? Including the node module changes, package.json, etc. We can discuss new commit hooks separately, not as part of this change.

File: examples/index.html (hunk 14 above)
Comment (johanbrandhorst) [reply examples/index.html]: You can also just remove it. I assume this has something to do with your npm init.

File: package.json (hunk 15 above)
Comment (johanbrandhorst) [reply package.json]: Please remove it.

File: scripts/install.sh (hunk 16 above)
Comment (johanbrandhorst) [reply scripts/install.sh]: OK, lets remove it please.

File: README.md (hunk 5 above)
Comment (johanbrandhorst) [reply README.md]: My comment doesn't contradict that. We should still add `paths=source_relative` as an option to the go_out command like so: 
```
--go_out=plugins=grpc,paths=source_relative:${DEST}
```

File: scripts/install.sh (hunk 16 above)
Comment (yiakwy) [reply scripts/install.sh]: copy it.

File: package.json (hunk 15 above)
Comment (yiakwy) [reply package.json]: copy it

File: README.md (hunk 5 above)
Comment (yiakwy) [reply README.md]: copy it.

File: CHANGELOG.md (hunk 1 above)
Comment (yiakwy) [reply CHANGELOG.md]: Roger. You are right that we can discuss commit hook in a different thread.
